fenicsxprecice/fenicsxprecice.py

```diff
@@ -198,36 +198,78 @@ def write_data(self, mesh_name, write_function):
         write_data = convert_fenicsx_to_precice(write_function, self._fenicsx_vertices[mesh_name].get_coordinates())
         self._participant.write_data(
             mesh_name,
-            self._config.get_write_data_name(mesh_name),
+            write_data_name,
             self._precice_vertex_ids[mesh_name],
             write_data
         )
 
+    def validate_function_space(self, function_objects, mesh_name, checkIfFunction):
+        """_summary_
+
+        Args:
+            functions (_type_): a dict of Function and FunctionSpace
+        ------
+        Returns: a bool if all function spaces are equal in given dict
+        """
+
+        # get first and extract function space
+        function_space = None
+        if function_objects is None:
+            return None
+        else:
+            f = list(function_objects.values())[0]
+            if checkIfFunction and isinstance(f, fem.Function):
+                function_space = f.function_space
+            # preCICE will use default zero values for initialization.
+            elif isinstance(f, fem.FunctionSpace):
+                function_space = f
+            elif f is None:
+                pass
+            else:
+                if checkIfFunction:
+                    raise Exception("A given object in {} is neither of type dolfinx.functions.function.Function or "
+                                    "dolfinx.functions.functionspace.FunctionSpace".format(mesh_name))
+                else:
+                    raise Exception(
+                        "A given object of {} is not of type dolfinx.functions.functionspace.FunctionSpace".format(mesh_name))
+
+        if function_space is None and len(function_objects) > 1:
+            raise Exception("Invalid argument provided: At least one write function space is defined as None,"
+                            "but the number of given write functions was {}."
+                            "If no write function want to be used on mesh {}, set"
+                            "the write function object array to None instead!".format(len(function_objects, mesh_name)))
+
+        for fun in function_objects:
+            func = function_objects[fun]
+            if checkIfFunction and isinstance(func, fem.Function):
+                assert func.function_space == function_space
+            elif isinstance(func, fem.FunctionSpace):
+                assert func == function_space
+            else:
+                if checkIfFunction:
+                    raise Exception("A given object in {} is neither of type dolfinx.functions.function.Function or "
+                                    "dolfinx.functions.functionspace.FunctionSpace".format(mesh_name))
+                else:
+                    raise Exception(
+                        "A given object of {} is not of type dolfinx.functions.functionspace.FunctionSpace".format(mesh_name))
+
+        return function_space
+
     def initialize(self, coupling_meshes):
```

Comment:
```suggestion
    def initialize(self, coupling_meshes: CouplingMesh):
```
Again: Maybe use typo annotations?